components/usage/pkg/db/workspace_instance_usage.go

```diff
@@ -83,9 +111,11 @@ func ListUsage(ctx context.Context, conn *gorm.DB, attributionId AttributionID,
 				// started before query range, still running
 				Or("startedAt <= ? AND (stoppedAt > ? OR stoppedAt IS NULL)", from, to),
 		).
+		Offset(int(offset)).
+		Limit(int(limit)).
 		Find(&usageRecords)
 	if result.Error != nil {
-		return nil, fmt.Errorf("failed to get usage records: %s", result.Error)
+		return nil, 0, 0, fmt.Errorf("failed to get usage records: %s", result.Error)
 	}
-	return usageRecords, nil
+	return usageRecords, count, totalCreditsUsed, nil
```

Comment:
Four return values is a bit much. It would be great to return 2 values (a struct of three plus the err).

Reply:
Yeah, it felt already dirty.